src/components/user-defined-sets/NamedSetManager.js

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable jsx-a11y/accessible-emoji */
+import React, { useState } from 'react';
+import * as Sets from './sets';
+
+export default function NamedSetManager(props) {
+  const {
+    sets,
+    name,
+    onUpdateSets = (msg) => {
+      console.warn(`onUpdateSets from NamedSetManager ${msg}`);
+    },
+  } = props;
+  const set = sets.namedSets[name];
+  const [isEditing, setIsEditing] = useState(false);
+  const [setName, setSetName] = useState(name);
+
+  const renameSet = () => {
+    setIsEditing(false);
+    onUpdateSets(Sets.renameNamedSet(sets, name, setName));
+  };
+
+  const deleteSet = () => {
+    setIsEditing(false);
+    onUpdateSets(Sets.deleteNamedSet(sets, name));
+  };
+
+  const updateCurrentSet = (nameToSelect) => {
+    onUpdateSets(Sets.setCurrentSet(sets, sets.namedSets[nameToSelect]));
+  };
+
+  return (
+    <React.Fragment>
```

Comment:
I think this contains just one `<tr>` at the top level, so the Fragment shouldn't be necessary?